packages/legend-application-studio/src/stores/EditorGraphState.ts

```diff
@@ -1106,26 +1202,20 @@ export class EditorGraphState {
         this.graphGenerationState.generatedEntities,
         this.editorStore.graphManagerState.generationsBuildState,
       );
-
-      /**
-       * Reprocess explorer tree which might still hold references to old graph
-       *
-       * FIXME: we allow this so the UX stays the same but this can cause memory leak
-       * we could consider doing this properly using node IDs
-       *
-       * @risk memory-leak
-       */
-      this.editorStore.explorerTreeState.reprocess();
-
-      // so that information is not dependent on the graph, but on the component itself, with IDs and such.
-      this.editorStore.openedEditorStates = openedEditorStates
-        .map((editorState) =>
-          this.editorStore.reprocessElementEditorState(editorState),
-        )
-        .filter(isNonNullable);
-      this.editorStore.setCurrentEditorState(
-        this.editorStore.findCurrentEditorState(currentEditorState),
-      );
+      this.updateExplorerTree();
+      if (!inTextMode) {
```

Comment:
Will this change once we have a text for each element. 
We should just have a function called 
like buildOpenedElementEditors, which will have an implementation for text mode and form mode?

Reply:
yeah sure